utlis/function.js

```diff
@@ -0,0 +1,29 @@
+function search(nameFrag, targetArray) {
+  // chicking array and string
+  if (!nameFrag || !targetArray.length) return null;
+
+  return targetArray.filter((obj) => obj.name.includes(nameFrag));
+}
+
+function addToCartList(id, targetArray, wantedArray) {
+  if (
+    typeof id == "undefined" ||
+    !targetArray.length ||
+    !Array.isArray(wantedArray)
+  )
+    return null;
+  let rArray = [...wantedArray];
+  targetArray.forEach((obj) => obj.id == id && rArray.push(obj));
+  return rArray;
+}
```

Comment:
I'm not sure about what is the purpose of this function, definitely you need some descriptive naming and some comments if you are doing some logic that you wouldn't expect any reader to get immediately or for you when you come back to your code after a certain amount of time, believe you'll thank yourself that time for leaving some comments.